addons/sale_timesheet/controllers/main.py

```diff
@@ -18,13 +18,41 @@ def plan(self, domain):
         }
 
     def _prepare_plan_values(self, domain):
+
+        def float_to_time(hour_number):
+            minutes = hour_number * 60
+            return _('%sh%s') % (str(int(minutes/60)), '%02d' % int(minutes % 60))
+
+        timesheet_lines = request.env['account.analytic.line'].search(domain)
+
         values = {
             'currency': request.env.user.company_id.currency_id,
-            'timesheet_lines': request.env['account.analytic.line'].search(domain),
+            'timesheet_lines': timesheet_lines,
             'domain': domain,
+            'float_to_time': float_to_time,
         }
         hour_rounding = request.env.ref('product.product_uom_hour').rounding
-        billable_types = ['non_billable', 'billable_time', 'billable_fixed']
+        billable_types = ['non_billable', 'non_billable_project', 'billable_time', 'billable_fixed']
+
+        # -- Stat Buttons
+        stat_buttons = []
+        stat_buttons.append({
```

Comment:
Small note: As it is inherited in project_ts_forecast_sale, why not extracting it to a small helper method creating those famous stat button ?